ddpui/utils/warehouse/client/postgres.py

```
@@ -63,14 +66,76 @@ def __init__(self, creds: dict):
+        """Dispose SQLAlchemy connections held by the engine pool."""
+        self.engine.dispose()
+        return True
+
+    def get_columnspec(self, schema: str, table_id: str) -> list[str]:
+        """Gets the column schema for this table."""
+        resultset = self.execute(
+            f"""SELECT column_name
+                FROM information_schema.columns
+                WHERE table_schema = '{schema}'
+                AND table_name = '{table_id}'
+            """
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)